### ISSUE TITLE
ability to change client fetchOptions dynamically 

### ISSUE THREAD

**Issue by bogdansoare** [andyrichardson/tipple#53]
Let's say you set an Authorization header based on a token stored in local storage
```js
const token = getToken();

const headers = { "content-type": "application/json" };
if (token) {
  headers.Authorization = `Bearer ${token}`;
}

const client = createClient({
  baseUrl: 'x',
  fetchOptions: {
    headers
  }
});
```

Now if I change or delete the token stored in local storage the client won't be updated with the new token. 
Maybe if the lib provided a function to change fetchOptions, something like apollo client does, would solve this case.

**Comment by andyrichardson** [comment]
Thanks for the suggestion. How about fetchOptions being optionally a function which returns a `RequestInit`?

**Comment by bogdansoare** [comment]
Yes, that would be great I think.

**Comment by jozan** [comment]
We have solved this by putting our Tipple client in context and recreate the client when session tokens change.

**Comment by andyrichardson** [comment]
@jozan nice idea! Heads up, you might be losing your cache state unless you use persistence.

This feature will get added within the next few days 👍

**Comment by andyrichardson** [comment]
Quick update folks, after looking into it, I realized I had [already implemented this in v0.4.2](https://github.com/andyrichardson/tipple/blob/master/src/client.ts#L8).

The docs can be [found here](https://github.com/andyrichardson/tipple/blob/master/docs/Configuration.md#fetchoptions). Let me know if this doesn't satisfy your use case.
